charts/ocis/templates/storageusers/secret.yaml

```diff
@@ -0,0 +1,15 @@
+{{ if eq .Values.services.storageusers.storageBackend.driver "s3ng" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.secretRefs.s3CredentialsSecretRef}}
+data:
+  {{- $secret := lookup "v1" "Secret" .Release.Namespace .Values.secretRefs.s3CredentialsSecretRef }}
```

Comment:
This makes for a slightly confusing behavior when first running `helmfile sync` and then `helmfile diff`. I added secret and key to my helmfile after first applying the config, and now when running the diff I always get:
```
$ helmfile diff
Building dependency release=ocis, chart=../../charts/ocis
Comparing release=redis, chart=bitnami/redis
Comparing release=ocis, chart=../../charts/ocis
ocis, s3-storage-users-credentials, Secret (v1) has changed:
  # Source: ocis/templates/storageusers/secret.yaml
  apiVersion: v1
  kind: Secret
  metadata:
    name: s3-storage-users-credentials
  data:
-   accessKey: '-------- # (11 bytes)'
-   secretKey: '-------- # (11 bytes)'
+   accessKey: '++++++++ # (5 bytes)'
+   secretKey: '++++++++ # (8 bytes)'
```
I think there should probably be a config option to create a secret _or_ use an existing secret.

Reply:
I'm not sure, I'd say that the vast majority of the users would never do a diff, so would never see this.

Reply:
I mean, I have no problem making this a config setting, but we already have so many, and if we do do that, we should make it a global one for all secrets. Furthermore, not clobbering secrets like this is quite common.

Reply:
Hmm. I kind of like the idea of having one global setting for getting secrets from config or expecting them to just exist.

That's an easily documented setting.

@wkloucek WDYT?

Reply:
Agreed, but then we'll have to convert all the other secrets to this method as well.

I personally think it that should be its own PR.

Reply:
> This makes for a slightly confusing behavior when first running `helmfile sync` and then `helmfile diff`. I added secret and key to my helmfile after first applying the config, and now when running the diff I always get:
> 
> ```
> $ helmfile diff
> Building dependency release=ocis, chart=../../charts/ocis
> Comparing release=redis, chart=bitnami/redis
> Comparing release=ocis, chart=../../charts/ocis
> ocis, s3-storage-users-credentials, Secret (v1) has changed:
>   # Source: ocis/templates/storageusers/secret.yaml
>   apiVersion: v1
>   kind: Secret
>   metadata:
>     name: s3-storage-users-credentials
>   data:
> -   accessKey: '-------- # (11 bytes)'
> -   secretKey: '-------- # (11 bytes)'
> +   accessKey: '++++++++ # (5 bytes)'
> +   secretKey: '++++++++ # (8 bytes)'
> ```
> 
> I think there should probably be a config option to create a secret _or_ use an existing secret.

I saw this behaviour in other charts (grafana) as well. So seems to be normal.

Reply:
I went from the other way around, as the current default is the keys being set, I didn't want to overwrite an already existing secret. If you're fine with me making those empty by default, I have no problems with changing it to the way you describe.

Reply:
In that case I will make that the behaviour.

Reply:
After a discussion with @wkloucek about helm deletion behaviour, I added a deprecation/best practices warning to this.